### DISCUSSION TITLE
Interpolation to Query Scalar Values at Arbitrary Points Within a Mesh Domain

### DISCUSSION THREAD

**Discussion by daalgi** [pyvista/pyvista#6603] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
import pyvista as pv
import numpy as np

points = np.array([
    [0, 0, 0], 
    [2, 0, 0], 
    [2, 2, 0], 
    [0, 2, 0], 
    [4, 0, 0], 
    [4, 3, 0],
])
cells = np.array([
    [4, 0, 1, 2, 3], 
    [4, 1, 4, 5, 2],
])
cell_types = np.array([pv.CellType.QUAD, pv.CellType.QUAD])
grid = pv.UnstructuredGrid(cells, cell_types, points)

nodal_values = np.array([1., 2., 2., 1., 3., 3.])
grid.point_data['Nodal Values'] = nodal_values

# Query points
points = [[x, 0, 0] for x in np.arange(0, 4.5, 0.5)]
poly_data = grid.extract_surface()
query_points = np.array(points)
query_poly_data = pv.PolyData(query_points)
interpolated_values = query_poly_data.interpolate(poly_data, sharpness=3)
print(interpolated_values.point_data["Nodal Values"])
# [1.  1.  1.5 2.  2.  2.  2.5 3.  3. ]
```

### Description

I’m using PyVista to interpolate scalar values at arbitrary points within a mesh. Currently, the results seem to only interpolate correctly if the point is at the midpoint between defined mesh points; otherwise, it just returns the value at the nearest mesh point.

Issue:
- Objective: Get interpolated scalar values at any point (x, y, z) within the mesh.
- Current Behavior: Returns scalar values at the nearest mesh point, not interpolated values, unless the point is precisely between defined mesh points.

### System Information

```shell
OS : Windows (10 10.0.19045 SP0 Multiprocessor Free)
              CPU(s) : 12
             Machine : AMD64
        Architecture : 64bit
                 RAM : 31.9 GiB
         Environment : Jupyter
          GPU Vendor : Intel
        GPU Renderer : Intel(R) UHD Graphics
         GPU Version : 4.5.0 - Build 31.0.101.2111
    MathText Support : False

  Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64
  bit (AMD64)]

             pyvista : 0.43.10
                 vtk : 9.3.0
               numpy : 1.26.4
          matplotlib : 3.8.4
              scooby : 0.10.0
               pooch : 1.8.2
              pillow : 10.3.0
...
       trame_vuetify : 2.6.0
jupyter_server_proxy : 4.2.0
        nest_asyncio : 1.6.0
```

**Comment by MatthewFlamm** [comment, marked as answer]
You want `sample` not `interpolate`, see https://docs.pyvista.org/examples/01-filter/resample#resampling-example.

**Reply by banesullivan** [reply]
I'm very confused by this (once again sample vs interpolate vs probe being confusing). But I think the linked example clarifies well, thanks @MatthewFlamm!

**Reply by MatthewFlamm** [reply]
To be clear, `probe` is no longer a filter.  It was a less useful version of `sample` previously.

`sample` is for 'intepolating' within cells.  `interpolate` is for 'interpolating' amongst nonconnected points.  I _think_ the example helps and similar language is repeated in documentation for each filter.  Maybe it is the naming of the filters themselves that need to change?

**Reply by banesullivan** [reply]
I think the naming here is inherited from the implementation in VTK which we ought to stick to

**Reply by MatthewFlamm** [reply]
The vtk name is `vtkPointInterpolator`, which helps a bit to remember that it is for interpolating from points not cells.  We could consider `interpolate_points` but it is longer and not obvious to me that it will be more understandable to new users.  Plus it will be an annoying breaking change.  I agree with you that we should stick with the naming.  Maybe we need a dedicated example that contrasts the two rather than having separate examples each with a monologue about the differences.

**Reply by banesullivan** [reply]
That's a great idea -- I think having an example overviewing both and calling out this confusing aspect will go a long way

**Reply by MatthewFlamm** [reply]
See #6639
